matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/DefaultCryptoService.kt

```diff
@@ -953,6 +957,7 @@ internal class DefaultCryptoService @Inject constructor(
         val eventContent = event.content.toModel<RoomHistoryVisibilityContent>()
         eventContent?.historyVisibility?.let {
             cryptoStore.setShouldEncryptForInvitedMembers(roomId, it != RoomHistoryVisibility.JOINED)
+            cryptoStore.setShouldShareHistory(roomId, it.shouldShareHistory())
```

Comment:
I'd a bit paranoid here and put this to false if eventContent?.historyVisibility is null